Update User Guide and Documentation: (#308)
AY2425S1-CS2103T-T14-4/tp: docs/UserGuide.md

## 4.4 Arguments

> Arguments are the values that are used in a command. **Arguments cannot be empty** and each must meet specific parsing and format requirements to ensure proper execution of the command.

Reviewer: Hmm, I personally prefer `follow each flag` since it suggests it comes right after flags. What does the rest of the team think?

Author: For this, I kept the initial phrasing for the arguments specifically for Flags (New section 4.1), so this amendment is for the general Arguments

Currently I split into Flag Arguments and Non-Flag arguments for clarity, albeit the only non-flag argument we currently have is Index, so I'm not too sure if this is the most appropriate. What do the rest think?

>
> Providing a flag without an accompanying argument will result in an error and prevent the command from executing properly.

### 4.4.2 Non-Flag Arguments 

Non-flag arguments are the values that do not have a corresponding tag. Currently, this only includes the `<INDEX>` argument.

Refer to the table below for more details.

| **Expected Argument** | **Description**                     | **Requirements**                                                                                             |
|-----------------------|-------------------------------------|--------------------------------------------------------------------------------------------------------------|
| `<INDEX>`             | The index of the item to be edited. | Positive number.<br/> • Must be less than or equal to the total number of clients in the current list view.  |

> 💡 **Pro Tip:**
> 

> > Indexes will when the current list view changes!

Reviewer: Just a typo!

Author: Thanks for the catch! Resolved

> 💡 **Pro Tip:**
> No need to worry about duplicate indexes—AgentAssist guarantees that every client has a unique index automatically.
- **Note:** If the value for `Tier`, `Status` or `REMARK` is the default value (`NA`), they will not be shown.

>   - This can be used to remove all remarks for a given client: `edit <INDEX> rn/NA` 

Reviewer 2: ```suggestion
  - This can be used to remove all remarks for a given client: `edit <INDEX> rn/NA` .
```

delete <INDEX>
```
* Mandatory Field: `<INDEX>`

> * Note: The provided `<INDEX>` must be **greater than 0 and less than the total number of clients in the list**.

Reviewer: Was this removed by accident? It seems quite important.

Author: This was centralised to the index arguments in Section 4, but I'll add a hyperlink as a quick reference!

Reviewer: I think a hyperlink sounds good, but I feel repeating this is ok since its just a 1 line reminder

Reviewer 2: Yes I second adding a hyperlink to the initial definition, to avoid repeating.

Reviewer: Maybe we can briefly write it and have it linked to the section with more details!

## 5.6 Modifying the Data File
The data in AgentAssist is automatically saved as a [JSON](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/JSON) file at `[JAR file location]/data/agentassist.json`. Advanced users are welcome to update data directly by editing that data file.

> ⚠️ **Danger:**
> If the data file format becomes invalid, AgentAssist will **discard all data** and start with an empty file on the next run. It's strongly recommended to back up the file before any manual edits.
>
> Incorrect data modifications may also cause unexpected behavior. Do take note that no warnings will be given. **Only modify the data file if you're confident in doing so correctly.** 

> ## 5.7 Modifying the Window Size

Reviewer: Might be better to name this `Modifying the Preferences File" and add similar warnings as in the above data section to avoid modifying unless the user knows what they are doing.

Author: Updated to reflect the changes!

2. **If you minimize the Help Window** and then run the `help` command (or use the `Help` menu, or the keyboard shortcut `F1`) again, the original Help Window will remain minimized, and no new Help Window will appear. The remedy is to manually restore the minimized Help Window.
3. **When inputting names that use `/`**, this can cause an `invalid command` error to be shown or even unintended attributes for the client added. Avoid using `/`, and spell out the names in full, for cases like 'Ramesh s/o Ravichandran', change it to 'Ramesh Son Of Ravichandran'
4.  **Using non-english text input can cause visual bugs**. This release fully supports English text input only. Using non-English characters or text—especially those with right-to-left direction, like Arabic—may result in display problems, including incorrect text alignment, direction, and character rendering. We are actively working to expand support for international languages in future releases.
5. **For index based commands**, negative inputs and inputs larger than 2147483647 result in inconsistent error messages. We are actively working on resolving this inconsistency in future releases.

> 6. **When using the edit command**, an edit that results in no change to the client is a considered valid command. We are actively working to expand the warnings to notify the user of this scenario in future releases.

Reviewer: ```suggestion
6. **For the edit command**, edits that do not result in changes to client details are currently considered valid. We are working on adding warnings to notify users of unchanged edits in future releases.
```

2. **If you minimize the Help Window** and then run the `help` command (or use the `Help` menu, or the keyboard shortcut `F1`) again, the original Help Window will remain minimized, and no new Help Window will appear. The remedy is to manually restore the minimized Help Window.

> 3. **When inputting names that use `/`**, this can cause an `invalid command` error to be shown or even unintended attributes for the client added. Avoid using `/`, and spell out the names in full, for cases like 'Ramesh s/o Ravichandran', change it to 'Ramesh Son Of Ravichandran'

Reviewer 2: ```suggestion
3. **When inputting names that use `/`**, this can cause an `invalid command` error to be shown or even unintended attributes for the client added. Avoid using `/`, and spell out the names in full, for cases like 'Ramesh s/o Ravichandran', change it to 'Ramesh Son Of Ravichandran'.
```
